### DISCUSSION TITLE
Filtering in `findMany` with JSONB = null does not work in NOT, AND & OR operator

### DISCUSSION THREAD

**Discussion by j0hannr** [prisma/prisma#18142] [Q&A] [answered]
### Filtering in `findMany` with JSONB = null does not work in NOT, AND & OR operator

I have got entries in the `property` table and want to filter by multiple empty fields. 

Unfortunately filtering for empty `JSONB` fields, does not return the fields where the JSON is `null` | `undefined`

```js
await prisma.property.findMany({
   where: {
      OR: [
         { geometry: undefined },
         { geometry: { equals: undefined } },
         { osmId: null }
      ]
   }
})
```

### Schema

```prisma

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Property {
  id           String           @id @default(cuid())
  street       String?
  postcode     String?
  country      String?
  verified     Boolean?         @default(false)
  public       Boolean?         @default(false)
  createdAt    DateTime         @default(now())
  updatedAt    DateTime         @default(now()) @updatedAt
  createdBy    String?
  city         String?
  osmId        Int?
  geometry     Json?
  region       String?
  streetNumber String?
  parent       String?
  lat          Float?
  lng          Float?
  type         PropertyType     @default(building)
  Device       Device[]
  Option       Option[]
  Parameter    Parameter[]
  User         UserOnProperty[]

  @@unique([street, streetNumber, city, postcode, country], name: "AddressIdentifier")
}
```

### Prisma Version
```
prisma                  : 4.11.0
@prisma/client          : 4.11.0
Current platform        : darwin
Query Engine (Node-API) : libquery-engine 8fde8fef4033376662cad983758335009d522acb (at node_modules/@prisma/engines/libquery_engine-darwin.dylib.node)
Migration Engine        : migration-engine-cli 8fde8fef4033376662cad983758335009d522acb (at node_modules/@prisma/engines/migration-engine-darwin)
Format Wasm             : @prisma/prisma-fmt-wasm 4.11.0-57.8fde8fef4033376662cad983758335009d522acb
Default Engines Hash    : 8fde8fef4033376662cad983758335009d522acb
Studio                  : 0.483.0
```

### Previous Issues / Discusions
https://github.com/prisma/prisma/discussions/8373
https://github.com/prisma/prisma/issues/8399
https://github.com/prisma/prisma/pull/9074/files

**Comment by nurul3101** [comment]
Hey @j0hannr 👋 

Can you try passing [Prisma.JsonNull](https://www.prisma.io/docs/guides/upgrade-guides/upgrading-versions/upgrading-to-prisma-3#json-null-equality), When you pass `undefined` as a value for a field, it would be equivalent of passing nothing at all.

So in your case, it would be equivalent of not passing geometry field in the where clause. Here's a [reference](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined) for this behaviour.

**Comment by j0hannr** [comment]
Hi @nurul3101 ☺️

I tried both `geometry: { equals: Prisma.JsonNull }` and `geometry: Prisma.JsonNull` and it does not change the output.

### Full query
```js
const buildingsToCrawl = await prisma.property.findMany({
    where: {
      type: "building",
      OR: [{ geometry: { equals: Prisma.JsonNull } }, { osmId: null }],
    },
    orderBy: {
      User: {
        _count: "desc",
      },
    },
    select: {
      id: true,
    },
  });
```

**Reply by nurul3101** [reply, marked as answer]
Does the output change with Prisma.AnyNull or Prisma.DbNull?

**Reply by j0hannr** [reply]
@nurul3101 Yes! Works with both 😌
Would be good to reference this in the documentation.

**Reply by nurul3101** [reply]
I am glad to hear it's working now 👍
